components/mpas-ocean/src/shared/mpas_ocn_redi.F

```diff
@@ -0,0 +1,443 @@
+        else
+           slope = sign(1.0_RKIND, den * num)
+        end if
+
+     end function
+
+!***********************************************************************
+
+   subroutine ocn_redi_compute_isopycnal_slopes(statePool, &
+                                                timeLevelIn)
+
+     !-----------------------------------------------------------------
+     !
+     ! input/output variables
+     !
+     !-----------------------------------------------------------------
+
+     type(mpas_pool_type), intent(in) :: &
+        statePool ! pool containing state variables
+
+     integer, intent(in), optional :: &
+                     timeLevelIn          ! time level for state variables
+
+     !-----------------------------------------------------------------
+     !
+     ! local variables
+     !
+     !-----------------------------------------------------------------
+
+     type(mpas_pool_type), pointer :: tracersPool
+     real(kind=RKIND), dimension(:, :, :), pointer :: activeTracers
+     integer, pointer :: indexTemperaturePtr, indexSalinityPtr
+     integer :: indexTemperature, indexSalinity, i, iCellSelf
+     real(kind=RKIND), dimension(:,:), allocatable :: dzTop, dTdzTop, dSdzTop
+     real(kind=RKIND), dimension(:), allocatable :: k33Norm
+
+     real(kind=RKIND), dimension(:,:), pointer :: layerThickness
+     real(kind=RKIND), dimension(:), pointer :: ssh
+
+     real(kind=RKIND) :: bldedge, mldedge, thickslope, scalefactor
+     real(kind=RKIND) :: dcEdgeInv, areaEdge, drhoDT, drhoDS, dTdx
+     real(kind=RKIND) :: dSdx, drhoDx, tempSlopeUp, tempSlopeDown
+     real(kind=RKIND) :: slopeTaperUp, slopeTaperDown, sfcTaper
+     real(kind=RKIND) :: zMLD, BruntVaisalaFreqTopEdge, maxN, sfcTaperTemp
+     integer :: cell1, cell2, iCell, iEdge, k, timeLevel, nCells, nEdges
+
+     real(kind=RKIND), parameter :: epsGM = 1.0e-12_RKIND
+     if (present(timeLevelIn)) then
+        timeLevel = timeLevelIn
+     else
+        timeLevel = 1
+     end if
+
+     call mpas_timer_start('redi compute slopes')
+
+     call mpas_pool_get_array(statePool, 'layerThickness', layerThickness, timeLevel)
+     call mpas_pool_get_subpool(statePool, 'tracers', tracersPool)
+     call mpas_pool_get_array(statePool, 'ssh', ssh, timeLevel)
+     call mpas_pool_get_array(tracersPool, 'activeTracers', activeTracers, timeLevel)
+     call mpas_pool_get_dimension(tracersPool, 'index_temperature', indexTemperaturePtr)
+     call mpas_pool_get_dimension(tracersPool, 'index_salinity', indexSalinityPtr)
+     indexTemperature = indexTemperaturePtr
+     indexSalinity    = indexSalinityPtr
+
+     !Initialize RediKappa to the constant coefficient value and modify if equalGM or N2_dependent
+     !$omp parallel
+     !$omp do schedule(runtime) private(k)
+     do iEdge = 1, nEdgesAll
+        do k = 1, nVertLevels
+           RediKappaSfcTaper(k,iEdge) = 1.0_RKIND
+           RediKappa(k,iEdge) = config_Redi_constant_kappa*RediHorizontalTaper(iEdge)
```

Comment:
@vanroekel -- this line also causes non-BFB behavior with master. There's nothing like it in that codebase